server/api/crud/alerts.py

```diff
@@ -0,0 +1,269 @@
+from server.api.utils.notification_pusher import AlertNotificationPusher
+
+
+class Alerts(
+    metaclass=mlrun.utils.singleton.Singleton,
+):
+    def store_alert(
+        self,
+        session: sqlalchemy.orm.Session,
+        project: str,
+        name: str,
+        alert_data: mlrun.common.schemas.AlertConfig,
+    ):
+        project = project or mlrun.mlconf.default_project
+
+        alert = server.api.utils.singletons.db.get_db().get_alert(
+            session, project, name
+        )
+
+        self._validate_alert(alert_data, name, project)
+
+        if alert is not None:
+            self._delete_notifications(alert)
+
+        alert_data.notifications = [
+            mlrun.common.schemas.notification.Notification(**x.to_dict())
+            for x in server.api.api.utils.validate_and_mask_notification_list(
+                alert_data.notifications, None, project
+            )
+        ]
+
+        if alert is not None:
+            for kind in alert.trigger.events:
+                server.api.crud.Events().remove_event(project, kind)
+            alert_data.created = alert.created
+            alert_data.id = alert.id
+            new_alert = server.api.utils.singletons.db.get_db().store_alert(
+                session, alert_data
+            )
+        else:
+            new_alert = server.api.utils.singletons.db.get_db().create_alert(
+                session, alert_data
+            )
+
+        for kind in new_alert.trigger.events:
+            server.api.crud.Events().add_event(project, kind, new_alert.name)
+
+        self.reset_alert(session, project, new_alert.name)
+
+        server.api.utils.singletons.db.get_db().enrich_alert(session, new_alert)
+
+        return new_alert
+
+    def list_alerts(
+        self,
+        session: sqlalchemy.orm.Session,
+        project: str = "",
+    ) -> list[mlrun.common.schemas.AlertConfig]:
+        project = project or mlrun.mlconf.default_project
+        return server.api.utils.singletons.db.get_db().list_alerts(session, project)
+
+    def get_enriched_alert(
+        self, session: sqlalchemy.orm.Session, project: str, name: str
+    ):
+        alert = server.api.utils.singletons.db.get_db().get_alert(
+            session, project, name
+        )
+        if alert is not None:
+            server.api.utils.singletons.db.get_db().enrich_alert(session, alert)
+        return alert
+
+    def get_alert(
+        self,
+        session: sqlalchemy.orm.Session,
+        project: str,
+        name: str,
+    ) -> mlrun.common.schemas.AlertConfig:
+        project = project or mlrun.mlconf.default_project
+        return server.api.utils.singletons.db.get_db().get_alert(session, project, name)
+
+    def get_alert_by_id(
+        self,
+        session: sqlalchemy.orm.Session,
+        alert_id: str,
+    ) -> mlrun.common.schemas.AlertConfig:
+        return server.api.utils.singletons.db.get_db().get_alert_by_id(
+            session, alert_id
+        )
```

Comment:
Is this used anywhere? Looks like this is internal functionality that is needed internally by the DB, but not used externally. Why do you expose it in the CRUD?